### PR TITLE
Add first step for proofs documentation

### PR DESCRIPTION
**Pull request by nafur**
This PR initializes some documentation for our proofs. It adds some almost empty files and makes sure that we can show documentation for the `PfRule` enum.

### CODE REVIEW

**Comment by nafur** [comment]
The idea of this PR is to get the whole thing started so that we can then split the work... @ajreynol @HanielB @Lachnitt

File: docs/proofs/output_alethe.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Proof format: Alethe
+====================
+
+Using the flag :ref:`--proof-format-mode=alethe <lbl-option-proof-format-mode>`, cvc5 can output proofs in the `Alethe proof format <https://verit.loria.fr/documentation/alethe-spec.pdf>`_.
+
+Additonally, the following flags should be used to produce proofs with a minimal number of holes and no term sharing (which is not supported for Alethe yet): :ref:`--simplification=none <lbl-option-simplification>` :ref:`--dag-thresh=0 <lbl-option-dag-thresh>` :ref:`--proof-granularity=dsl-rewrite <lbl-option-proof-granularity>`.
```
Comment (ajreynol) [inline docs/proofs/output_alethe.rst:6]: `dsl-rewrite` will not be supported in the relase.  It is only available on `proof-new`.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: docs/proofs/output_alethe.rst (hunk 1 above)
Comment (nafur) [reply docs/proofs/output_alethe.rst:6]: So what should it say instead? (I don't actually feel responsible for the content :) )

Comment (ajreynol) [reply docs/proofs/output_alethe.rst:6]: @Lachnitt

Comment (Lachnitt) [reply docs/proofs/output_alethe.rst:6]: I will just delete that part then. All THEORY_REWRITE will just be printed as all_simplifies which is fine.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**